test/Microsoft.SqlTools.Dmp.Hosting.UnitTests/ExtensibilityTests/ServiceProviderTests.cs

```diff
@@ -0,0 +1,116 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Linq;
+using Microsoft.SqlTools.Dmp.Hosting.Extensibility;
+using Microsoft.SqlTools.Dmp.Hosting.Utility;
+using Xunit;
+
+namespace Microsoft.SqlTools.Dmp.Hosting.UnitTests.ExtensibilityTests
+{
+    public class ServiceProviderTests
+    {
+        private readonly RegisteredServiceProvider provider;
+        public ServiceProviderTests()
+        {
+            provider = new RegisteredServiceProvider();
+        }
+
+        [Fact]
+        public void GetServiceShouldReturnNullIfNoServicesRegistered()
+        {
+            // Given no service registered
+            // When I call GetService
+            var service = provider.GetService<MyProviderService>();
+            // Then I expect null to be returned
+            Assert.Null(service);
+        }
+
+
+        [Fact]
+        public void GetSingleServiceThrowsMultipleServicesRegistered()
+        {
+            // Given 2 services registered
+            provider.Register(() => new[] { new MyProviderService(), new MyProviderService() });
+            // When I call GetService
+            // Then I expect to throw
+            Assert.Throws<InvalidOperationException>(() => provider.GetService<MyProviderService>());
+        }
+
+        [Fact]
+        public void GetServicesShouldReturnEmptyIfNoServicesRegistered()
+        {
+            // Given no service regisstered
+            // When I call GetService
+            var services = provider.GetServices<MyProviderService>();
+            // Then I expect empty enumerable to be returned
+            Assert.NotNull(services);
+            Assert.Equal(0, services.Count());
+        }
+
+        [Fact]
+        public void GetServiceShouldReturnRegisteredService()
+        {
+            MyProviderService service = new MyProviderService();
+            provider.RegisterSingleService(service);
+
+            var returnedService = provider.GetService<MyProviderService>();
+            Assert.Equal(service, returnedService);            
+        }
+
+        [Fact]
+        public void GetServicesShouldReturnRegisteredServiceWhenMultipleServicesRegistered()
+        {
+            MyProviderService service = new MyProviderService();
+            provider.RegisterSingleService(service);          
+
+            var returnedServices = provider.GetServices<MyProviderService>();
+            Assert.Equal(service, returnedServices.Single());
+        }
+
+//        [Fact]
```

Comment:
Were these always commented out? If so, delete. If not, when will we be re-enabling / why are they commented out?

Reply:
They were not commented out originally. The issue was that they had references to ASTNodeFormatterFactory (some random exported type from the tools service). That reference isn't good anymore with this refactor. So, at some point we need to reenable with unit test safe references.